Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/cherry_pick.py

```diff
@@ -89,7 +89,14 @@ def main(cls, args, repository, **kwargs):
 
         env = os.environ.copy()
         env['GIT_WEBKIT_CHERRY_PICKED'] = cherry_pick_string
-        env['GIT_WEBKIT_BUG'] = issue.link if issue else ''
+        env['COMMIT_MESSAGE_BUG'] = issue.link if issue else ''
+
+        repository.branch_environment.cleanup()
+        repository.branch_environment.set(dict(
```

Comment:
Is it necessary to do the git-config dance when cherry picking? The cherry-pick should either succeed or fail with conflicts, and in the latter case the prepared commit message should be saved.

Reply:
TLDR, yes.

The most common cherry-picking pattern is defining a different radar (probably a clone of the original) and tracking the cherry-pick with that radar. In that case, we'd be defining a different "COMMIT_MESSAGE_BUG" than the original commit message has. If we were interrupted by a conflict resolution and didn't cache that value somehow, we would lose it. This is particularly relevant for combining this change with #12058. We will likely have a `--clone` option on our `cherry-pick` command that will do the cherry-picking and cloning in a single script invocation.

Reply:
> If we were interrupted by a conflict resolution and didn't cache that value somehow, we would lose it.

OK, makes sense. I didn't realize that during conflict resolution, .git/COMMIT_EDITMSG is the _original_ commit message, and `prepare-commit-msg` is not called until conflicts are fixed and you run `git cherry-pick --continue`.